src/application/snapshot/commands/get-proposal.ts

```diff
@@ -51,18 +59,24 @@ export class GetProposalCommand extends Command<Payload, ProposalData> {
         throw new HandlerError('Schema validation failed');
       }
       const proposal = validationResult.data.data.proposals[0];
-      if (!proposal) {
-        throw new HandlerError('Proposal not found');
+      let proposalWithResolvedAddress = null;
+
+      if (proposal) {
+        const resolvedProposalAddress = await resolveAddress(proposal.author);
+
+        proposalWithResolvedAddress = {
+          ...proposal,
+          author: {
+            address: proposal.author,
+            resolvedProposalAddress,
+          },
+        };
       }
 
-      const resolvedAddress = await resolveAddress(proposal.author);
       return new OkResult({
-        ...proposal,
-        author: {
-          address: proposal.author,
-          resolvedAddress,
-        },
-      });
+        proposal: proposalWithResolvedAddress,
+        hasNextProposal: !!validationResult.data.data.proposals[1],
+      } as Response);
```

Comment:
Maybe we could simplify it a little bit:
```ts
const [currentProposal, nextProposal] = validationResult.data.data.proposals;
const hasNextProposal = Boolean(nextProposal);
if(!current){
  return new OkResult({
    proposal: null,
    hasNextProposal: hasNextProposal, // probably should be hardcoded to false?
  })
}

const resolvedAddress = await resolveAddress(proposal.author);
return new OkResult({
    proposal: {
      ...proposal,
      author: { ...proposal.author, resolvedAddress } },
      hasNextProposal
  })
```

Reply:
Nice, I like this approach 👌